golem-worker-executor/src/services/projects.rs

```diff
@@ -0,0 +1,304 @@
+                                    project_id: Some(project_id.clone().into()),
+                                },
+                                access_token,
+                            );
+                            Box::pin(client.get_project(request))
+                        })
+                        .await?
+                        .into_inner();
+
+                    match response
+                        .result
+                        .expect("Didn't receive expected field result")
+                    {
+                        get_project_response::Result::Success(payload) => Ok(payload
+                            .data
+                            .expect("didn't receive expected owner_account_id")
+                            .owner_account_id
+                            .expect("failed to receive project owner_account_id")
+                            .into()),
+                        get_project_response::Result::Error(err) => Err(GrpcError::Domain(err))?,
+                    }
+                })
+            },
+            is_grpc_retriable::<ProjectError>,
+        )
+        .await
+        .map_err(|err| WorkerExecutorError::unknown(format!("Failed to get project owner: {err}")))
+    }
+}
+
+#[async_trait]
+impl ProjectService for ProjectServiceGrpc {
+    async fn get_project_owner(
+        &self,
+        project_id: &ProjectId,
+    ) -> Result<AccountId, WorkerExecutorError> {
+        self.project_owner_cache
+            .get_or_insert_simple(project_id, || {
+                Box::pin(self.get_project_owner_remotely(project_id))
+            })
+            .await
+            .map_err(|e| WorkerExecutorError::unknown(format!("Failed to get project owner: {e}")))
+    }
+
+    async fn resolve_project(
+        &self,
+        account_id: &AccountId,
+        project_name: &str,
+    ) -> Result<Option<ProjectId>, WorkerExecutorError> {
+        self.resolved_project_cache
+            .get_or_insert_simple(&(account_id.clone(), project_name.to_string()), || {
+                Box::pin(self.resolve_project_remotely(account_id, project_name))
+            })
+            .await
+    }
+}
+
+fn create_cache<K, V>(
+    max_capacity: usize,
+    time_to_idle: Duration,
+) -> Cache<K, (), V, WorkerExecutorError>
+where
+    K: Eq + Hash + Clone + Send + Sync + 'static,
+    V: Clone + Send + Sync + 'static,
+{
+    Cache::new(
+        Some(max_capacity),
+        FullCacheEvictionMode::LeastRecentlyUsed(1),
+        BackgroundEvictionMode::OlderThan {
+            ttl: time_to_idle,
+            period: Duration::from_secs(60),
+        },
+        "resolved_project",
+    )
+}
+
+struct ProjectServiceDisabled {
+    account_id: AccountId,
+    project_id: ProjectId,
+}
+
+#[async_trait]
+impl ProjectService for ProjectServiceDisabled {
+    async fn get_project_owner(
+        &self,
+        _project_id: &ProjectId,
+    ) -> Result<AccountId, WorkerExecutorError> {
+        Ok(self.account_id.clone())
```

Comment:
Let's assert here that the account id is as expected, just to detect misconfigurations earlier instead of continuing with possibly wrong data